.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,61 @@
+# Contributing
+
+Please read EasyPost's general [contribution guidelines](https://github.com/EasyPost/.github/blob/main/CONTRIBUTING.md)
+
+## Code of conduct
+
+Please read EasyPost's [code of conduct](https://github.com/EasyPost/.github/blob/main/CODE_OF_CONDUCT.md)
+
+## Component API best practices
+
+If your pull request involves changes to any component APIs, please read over our list of best practices below.
+
+### One-offs
+
+We define a one-off as a deviation from best practices. We understand that one-offs are virtually inevitable, so when we create them, we should have a good reason for doing so as creating one-offs adds additional context for contributors and users as well as introduces the possibility of creating inconsistent UX.
+
+### Intuitive naming
+
+Property names should be descriptive and intuitive of what they do. Intuitive names are preferred over descriptive names. For example, for a button, “urlLinkDestination” is a descriptive name, but “href” is intuitive.
+
+### Consistent naming
+
+Properties that have identical purposes across different components should share the same name. Similarly, properties that have identical purposes should also share the same type. If one component has a boolean “isValid”, other components that intend to control validity should do the same.
+
+### Property dependency
+
+Although unavoidable at times, we should do our best to not create properties that are dependent upon another. This helps avoid configurations that lead to invalid states. If a component has multiple dependent properties, take a closer look at the overall structure of the component.
+
+### Booleans
+
+- Booleans should be prefixed with simple verbs. For a button, the property name “isActive” is preferred to “active”.
+- By default booleans should be false. This makes the notation a bit cleaner for when the boolean is active:
+
+  ```JSX
+    // avoid
+   <Button isNotDisabled={false} />
+  ```
+
+  ```JSX
+  // preferred
+  <Button isDisabled />
+  ```
+
+- Avoid creating booleans with hierarchical structure. If an icon component can have more than two possible placements, avoid “isIconStart”, “isIconMiddle”, and “isIconEnd” since we’d have to decide which variant takes precedence. Instead use an enum.
+
+### Enums
+
+- If an enum has only two values, a boolean should be used instead
```

Comment:
i'm not sure i agree with this one in an absolute sense haha. i see them as being slightly semantically different, and i think there are cases where you may want to enumerate only two values. is there a way we could soften the wording to be like "If an enum has only two values, consider if a boolean would make more sense."?

Reply:
thats a good point and i think im going to remove this part entirely. i got the idea from MUI as they have it in their docs:

![Screen Shot 2023-02-08 at 11 23 03 AM](https://user-images.githubusercontent.com/44451175/217631045-d2505671-5932-4df2-9fd3-340bd7247afc.png)

but they also have a couple instances of not really adhering to this and if i remember correctly, react-aria has some cases where they enumerate over two values as well. of course, we can get around those cases, but at that point we'd really be doing it for aesthetics as opposed to a functional purpose. i think as this library matures we can possibly comeback to adding a bit about it with the notable exceptions.

Reply:
those are great points. that sounds good to me, we can revisit as we better understand how we'd like to do it.

i actually think the second half of the enum statement from MUI sums up my thoughts on it well `or if there is the possibility that additional possible values may be required in the future`. basically how i see it is if the total potential domain is two a boolean makes sense, whereas if the total potential domain is more than 2, even if it currently is 2, likely an enum makes more sense.